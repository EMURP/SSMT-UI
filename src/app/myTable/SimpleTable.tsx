import React from "react";

import ReactDOM from 'react-dom';
import "@patternfly/react-core/dist/styles/base.css";
//import './fonts.css';

import {
  Table,
  TableHeader,
  TableBody,
  textCenter,
} from '@patternfly/react-table';

type myProps={}

type myState = {
    columns: Array<object | string >;
    rows: Array<object >;

}

const textFormatting = {

    transforms: [textCenter],
    cellTransforms: [textCenter]

} as React.CSSProperties

class SimpleTable extends React.Component<myProps,myState> {
  constructor(myProps) {
    super(myProps);
    this.state = {

      columns: [
                    { 
                        title: <div style={textFormatting}>Sr No</div>, 
                    },
                    {
                        title: <div style={textFormatting}> Project Name</div>,
                    },
                    {
                        title: <div style={textFormatting}> No. of Nodes </div>,
                    },
               
                ],

      rows:     [
                    {
                        cells: ['1', 'AI COE', '37', ]
                    },
                    {
                        cells: ['2', 'Teckton', '19']
                    },
                    {
                        cells: ['3', 'Chris', '64']
                    },
   
                ]

     
    };
}
       
 
  render() {
    // const { columns, rows } = this.state;
    return (
      <Table aria-label="Simple Table" cells={this.state.columns} rows={this.state.rows}>
        <TableHeader />
        <TableBody />
      </Table>
    );
  }
}
export { SimpleTable };