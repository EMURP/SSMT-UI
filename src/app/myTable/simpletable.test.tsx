import React from 'react';
import { shallow } from 'enzyme';
import { SimpleTable } from './SimpleTable';

describe('SimpleTable tests', () => {

  test('should render default SimpleTable component', () => {
    const view = shallow(<SimpleTable />);
    expect(view).toMatchSnapshot();
  })

})