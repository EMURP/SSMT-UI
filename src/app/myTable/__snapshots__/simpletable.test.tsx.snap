// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleTable tests should render default SimpleTable component 1`] = `
<Component
  aria-label="Simple Table"
  cells={
    Array [
      Object {
        "title": <div
          style={
            Object {
              "cellTransforms": Array [
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            }
          }
        >
          Sr No
        </div>,
      },
      Object {
        "title": <div
          style={
            Object {
              "cellTransforms": Array [
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            }
          }
        >
           Project Name
        </div>,
      },
      Object {
        "title": <div
          style={
            Object {
              "cellTransforms": Array [
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            }
          }
        >
           No. of Nodes 
        </div>,
      },
    ]
  }
  rows={
    Array [
      Object {
        "cells": Array [
          "1",
          "AI COE",
          "37",
        ],
      },
      Object {
        "cells": Array [
          "2",
          "Teckton",
          "19",
        ],
      },
      Object {
        "cells": Array [
          "3",
          "Chris",
          "64",
        ],
      },
    ]
  }
>
  <TableHeader />
  <TableBody />
</Component>
`;
